### ISSUE TITLE
tfe_workspace_ids data source returns unexpected workspaces

### ISSUE THREAD

**Issue by michael-tracey** [hashicorp/terraform-provider-tfe#747]
#### Terraform Cloud/Enterprise version

TFC for Business

#### Terraform version

```1.1.4```

#### Terraform Configuration Files

```
data "tfe_workspace_ids" "workspace_id" {
  names        = ["workspace_name_without_asterisks"]
  organization = var.tf_organization
}
```

#### Expected Behavior

If not using prefix or suffix asterisks * with the **tfe_workspace_ids** data source, matching should be a string comparison over the current string.Contains.  

#### Actual Behavior

This causes unexpected workspaces to get returned with the data source.  If your organization has workspaces with similar names, this data source may select workspaces belonging to different environments.

#### Additional Context

The issue is found here:  https://github.com/hashicorp/terraform-provider-tfe/blob/317355beefd74ab5ef618a6b47af7d4ed224eb66/tfe/data_source_workspace_ids.go#LL59 where strings without asterisks are processed, and then are still being matched with .Contains

**Comment by brandonc** [comment]
Thanks for tracking this down. We'll get this fixed and a new release ASAP.

**Comment by brandonc** [comment]
The fix was released in [v0.41.0](https://github.com/hashicorp/terraform-provider-tfe/blob/main/CHANGELOG.md#v0410-january-4-2023)
